hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.bootstrap;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.config.AdvancedNetworkConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.EndpointConfig;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.ServerSocketEndpointConfig;
+import com.hazelcast.config.alto.AltoSocketConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.instance.EndpointQualifier;
+import com.hazelcast.internal.tpc.AsyncServerSocket;
+import com.hazelcast.internal.tpc.AsyncSocket;
+import com.hazelcast.internal.tpc.Configuration;
+import com.hazelcast.internal.tpc.Reactor;
+import com.hazelcast.internal.tpc.ReadHandler;
+import com.hazelcast.internal.tpc.TpcEngine;
+import com.hazelcast.internal.tpc.nio.NioReactorBuilder;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationexecutor.impl.AltoOperationScheduler;
+import com.hazelcast.spi.impl.operationexecutor.impl.AltoPartitionOperationThread;
+import com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl;
+import com.hazelcast.spi.properties.HazelcastProperty;
+
+import java.io.UncheckedIOException;
+import java.net.BindException;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.internal.server.ServerContext.KILO_BYTE;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_KEEPALIVE;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_RCVBUF;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_REUSEPORT;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_SNDBUF;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.TCP_NODELAY;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@SuppressWarnings("checkstyle:MagicNumber, checkstyle:")
+public class TpcServerBootstrap {
+    public static final HazelcastProperty ALTO_ENABLED = new HazelcastProperty(
```

Comment:
since there is dedicated `AltoConfig` that also defines `enabled` & `eventLoopCount` properties, a comment here would be welcome to indicate these properties' values override the config.

Reply:
I will address this after merge under https://github.com/hazelcast/hazelcast/issues/23775.